### DISCUSSION TITLE
where is output frequency set for gocart history files?

### DISCUSSION THREAD

**Discussion by DeniseWorthen** [ufs-community/ufs-weather-model#1145] [Q&A] [answered]
In parm/gocart/AERO_HIST.rc, I see the following setting:

```
#
# Radiation-related diagnostics
#
  inst_aod.format:     'CFIO' ,
  inst_aod.template:   '%y4%m2%d2_%h2%n2z.nc4' ,
  inst_aod.archive:    '%c/Y%y4' ,
  inst_aod.mode:       'instantaneous'
  inst_aod.frequency:  010000,
```

Running the cpld_control_p8 test, the file in the run directory contains

```
#
# Radiation-related diagnostics
#
  inst_aod.format:     'CFIO' ,
  inst_aod.template:   '%y4%m2%d2_%h2%n2z.nc4' ,
  inst_aod.archive:    '%c/Y%y4' ,
  inst_aod.mode:       'instantaneous'
  inst_aod.frequency:  060000,
```

I believe the ``inst_aod.frequency`` controls the output for gocart history files in the RTs. Where is this value changed from '01' to '06'?

@lipan-NOAA Can you help answer this question?

**Comment by lipan-NOAA** [comment]
@DeniseWorthen You are right. "inst_aod.frequency" controls GOCART output frequency.

**Comment by DeniseWorthen** [comment]
Thanks---do you know how it gets set to "06" from the initial value of "01"? Does it somehow get changed using the FHOUT variable? I don't see any script file that modifies the AERO_HIST file.

**Comment by lipan-NOAA** [comment]
"AERO_HISTORY.rc" is the input file that controls GOCART output.  it is copied from ${CFGDIR} or ${CHM_CFGDIR} directory.

**Comment by DeniseWorthen** [comment]
I see what is happening now. In the UWM regression tests for the coupled model (tests/fv3_conf/cpld_control_run.IN), the file is actually being copied from the input-data directory, not the tests/parm directory:

```
#prognostic aerosols
if [ $CPLCHM = .true. ]; then
  cp      @[INPUTDATA_ROOT]/GOCART/p8/rc/*.rc .
  ln -sf  @[INPUTDATA_ROOT]/GOCART/p8/ExtData .
fi
```

The problem is that I need to shorten the length (the integration time) for the some of the regression tests using Aerosols + Waves in order to fit in the wall clock time of 30mins that we set for any regression test.

For example, the debug test needs to run for only fhmax=3 instead of fhmax=6. That means if I want to compare the gocart history file, I need to be able to write out the history at hour 3.

I need to do the same for the bmark_p8 Regression test, since it also runs for 6 hours, but I'm shortening the test to only 4 hours.

**Comment by junwang-noaa** [comment]
I think we  may use the parm/gocart/AERO_HIST.rc as a template so that we can change the GOCART output frequency for different tests.

**Comment by ShawnCebulaNOAA** [comment]
Hello @DeniseWorthen 

Were you able to get this figured out?  Thanks!
Shawn Cebula - EPIC End User Support

**Comment by lipan-NOAA** [comment, marked as answer]
GOCART outputs aerosol concentration fields and AOD. For AERO_HIST.rc, it control AOD output frequency; The aerosols concentration filed is in the file of atmfxxx.nc, which is controled by FHOUT.
